x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -72,11 +73,16 @@ public void clear() {
     // Set of completed sequences - separate to avoid polluting the other stages
     // It is a set since matches are ordered at insertion time based on the ordinal of the first entry
     private final Set<Sequence> completed;
+    private final Set<Sequence> toCheckForMissing;
     private final long maxSpanInNanos;
 
     private final boolean descending;
 
     private final Limit limit;
+    private final boolean[] missingEvents;
```

Comment:
Since the algorithm is complex enough, a bit more descriptive naming can help.
I would suggest a rename for this to `missingEventStages` (singular `event` and plural `stage`). In essence we are not talking about the actual events but about "stages" or "queries" or "criteria" (I am open to suggestions on which of these three words describe best this concept) that refer to matching missing events.